packages/react-tinacms-github/src/github-client/GithubFile.ts

```diff
@@ -0,0 +1,83 @@
+/**
+
+Copyright 2019 Forestry.io Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import { useMemo } from 'react'
+import { useCMS } from '@tinacms/react-core'
+
+type parseFn = (content: string) => any
+type serializeFn = (data: any) => string
+
+export class GithubFile {
+  private sha: string | null = null
+
+  constructor(
+    private cms: any,
+    private path: string,
+    private parse?: parseFn,
+    private serialize?: serializeFn
+  ) {}
+
+  fetchFile = async () => {
+    const res = await this.cms.api.github.fetchFile(this.path)
+    this.sha = res.sha
+    return this.parse ? this.parse(res.content) : res.content
+  }
+
+  commit = async (
+    data: any,
+    message: string = 'Update from TinaCMS',
+    retryOnConflict = true
+  ) => {
+    const serializedContent = this.serialize ? this.serialize(data) : data
+    try {
+      const response = await this.cms.api.github.commit(
+        this.path,
+        this.sha,
+        serializedContent,
+        message
+      )
+      this.sha = response.content.sha
+      this.cms.alerts.success(
+        `Saved Successfully: Changes committed to ${this.cms.api.github.workingRepoFullName}`
+      )
```

Comment:
I don't think this belongs here. By putting the alert in the github client we're preventing people from choosing the message themselves. This message would probably be fine for most English speakers, but i18n is the main concern. If someone from Italy is using Tina they can't change the text to Italian. 

I would prefer to push this to the client. If they want an alert, then instead of doing:

```js
useForm({
  onSave: file.commit,
})
```
they would do
```js
useForm({
  onSubmit: data => file.commit(data).then(() => cms.alert.success("Bueno!")
})
```

Reply:
changed it to fire an event instead of an alert. Developers then have the option to add the alert in `onSubmit` or just subscribe to all commit events

```
  cms.events.subscribe('github:commit', (response) => {
    cms.alerts.success('Changes saved successfully.')
  })
```

Reply:
I'm fine with this change, but I would also note here that there are lots of hard-coded console and alert messages elsewhere in Tina. If we want to tackle i18n holistically we'll want to think about some kind of translation solution; I don't think we should put the onus on Tina consumers to wire up alerts for every message that comes from the system

Reply:
>  I don't think we should put the onus on Tina consumers to wire up alerts for every message that comes from the system

I somewhat agree. I think we'll eventually get there, but It's something we should be slow to put an official stamp on until we see all the ways i18n will bite us.

I like you're idea of using plain events and triggering alerts from somewhere's else. My gut says that would make it easier to deal with i18n, because there could likely be a single source of truth for those messages. i.e.

```js
cms.events.subscribe("*", dispatchAlerts(cms))
```

And then maybe you can override:

```js
import { alerts: githubAlerts } from "react-tinacms-github"

cms.events.subscribe('*', alertDispatcher(cms, {
  ...githubAlerts,
  "event:name": { status: "error", message: "It broke", timeout: 4000 },
})
```

Where the function would look something like:

```js
interface Alerts {
  [type: string]: {
    message: string
    status: AlertStatus
    timeout?: number
  }
}

function alertDispatcher(cms: CMS, alerts: Alerts) {
  return (event) => {
    const alert = alerts[event.type]

    if (alert). cms.alerts.dispatch(alert)
  }
}
```